cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -50,15 +57,36 @@ FileObject<LogCollection> LogFileManager::readBatch(
     Aws::String aws_line(line.c_str());
     Aws::Utils::Json::JsonValue value(aws_line);
     Aws::CloudWatchLogs::Model::InputLogEvent input_event(value);
-    actual_batch_size++;
-    log_set.insert(input_event);
-    data_tokens.push_back(data_token);
+    pq.push(std::make_tuple(input_event.GetTimestamp(), line, data_token));
```

Comment:
It looks like the test class is a full copy-paste of this function, which won't be maintainable. I think all you need to do it override the `read` function from the superclass, and then this `readBatch` function can be tested directly, consuming fake data from it